File: LinkedList/InsertNodeAtPosition.js
//Insert Node At a specific Position

function Node(data, next) {
    this.data = data;
    this.next = next;
}

function LinkedList(currNode) {
    this.currNode = null;
}

LinkedList.prototype = {
    addNode: function(data) {
        var newNode = new Node(data, this.currNode);
        this.currNode = newNode;
    },
    addNodeAtPos: function(data, head, pos) {
        if (head == null || pos == 0) {
            return new Node(data, head);
        } else {
            var currPos = 1
            var newNode = head
            while (currPos < pos && newNode.next != null) {
                newNode = newNode.next
                currPos++;
            }
            newNode.next = new Node(data, newNode.next)
            return head

        }
    }
}

File: LinkedList/InsertNodeAtPosition.cs
//Insert Node At a specific Position

using System;

namespace ConsoleApplication1
{
    class Nodes
    {
        public int data;
        public Nodes nextnode;
        public Nodes(int _data, Nodes _nextnode)
        {
            this.data = _data;
            this.nextnode = _nextnode;
        }
    }
    class LinkedList
    {
        Nodes currnode;

        public LinkedList()
        {
            this.currnode = null;
        }

        void Add(int data)
        {
            Nodes _nodes = new Nodes(data, this.currnode);
            this.currnode = _nodes;
        }

        Nodes InsertNodeAtPos(int data,Nodes head,int Pos)
        {
            if(head == null || Pos == 0)
            {
                return new Nodes(data, head);
            }
            else
            {
                int currPos = 1;
                var newNode = head;
                while(newNode.nextnode !=null && currPos  < Pos)
                {
                    newNode = newNode.nextnode;
                    currPos++;
                }

                currnode.nextnode = new Nodes(data, currnode.nextnode);

                return head;
            }
        }
    }
}
